### ISSUE TITLE
Circular dependency shown in Ember app after installing some addons

### ISSUE THREAD

**Issue by amusedberry** [broccolijs/broccoli#389]
Hi. This is the first place I thought I should go, since the message does not have any additional information. I'm not an expert in node packages nor dependencies, but I'm happy to help if there's anything I can do!

# Problem

Circular dependency after running `ember serve` on a fresh installation with some add-ons. The building process is successful after that and the app runs as usual.

## Console error:

```shell
Circular dependency:
C:\Users\user\AppData\Local\Temp\broccoli-14236NwY61Erje7b6\cache-165-rollup\build\-private\system\model\model.js ->
C:\Users\user\AppData\Local\Temp\broccoli-14236NwY61Erje7b6\cache-165-rollup\build\-private\system\model\internal-model.js ->
C:\Users\user\AppData\Local\Temp\broccoli-14236NwY61Erje7b6\cache-165-rollup\build\-private\system\references.js ->
C:\Users\user\AppData\Local\Temp\broccoli-14236NwY61Erje7b6\cache-165-rollup\build\-private\system\references\belongs-to.js ->
C:\Users\user\AppData\Local\Temp\broccoli-14236NwY61Erje7b6\cache-165-rollup\build\-private\system\model\model.js
```

## Project package:

```js
{
  "name": "project",
  "version": "0.0.0",
  "private": true,
  "description": "Small description for front goes here",
  "repository": "",
  "license": "MIT",
  "author": "",
  "directories": {
    "doc": "doc",
    "test": "tests"
  },
  "scripts": {
    "build": "ember build",
    "lint:hbs": "ember-template-lint .",
    "lint:js": "eslint .",
    "start": "ember serve",
    "test": "ember test"
  },
  "devDependencies": {
    "@ember/jquery": "^0.5.2",
    "@ember/optional-features": "^0.6.3",
    "broccoli-asset-rev": "^2.7.0",
    "ember-ajax": "^3.1.0",
    "ember-cli": "~3.7.1",
    "ember-cli-app-version": "^3.2.0",
    "ember-cli-babel": "^7.1.2",
    "ember-cli-dependency-checker": "^3.0.0",
    "ember-cli-eslint": "^4.2.3",
    "ember-cli-htmlbars": "^3.0.0",
    "ember-cli-htmlbars-inline-precompile": "^1.0.3",
    "ember-cli-inject-live-reload": "^1.8.2",
    "ember-cli-mirage": "^0.4.15",
    "ember-cli-moment-shim": "^3.7.1",
    "ember-cli-sass": "^10.0.0",
    "ember-cli-sri": "^2.1.1",
    "ember-cli-template-lint": "^1.0.0-beta.1",
    "ember-cli-uglify": "^2.1.0",
    "ember-concurrency": "^0.8.27",
    "ember-data": "~3.7.0",
    "ember-export-application-global": "^2.0.0",
    "ember-fetch": "^6.4.0",
    "ember-load-initializers": "^1.1.0",
    "ember-maybe-import-regenerator": "^0.1.6",
    "ember-moment": "^7.8.1",
    "ember-qunit": "^3.4.1",
    "ember-resolver": "^5.0.1",
    "ember-simple-auth": "^1.8.1",
    "ember-source": "~3.7.0",
    "eslint-plugin-ember": "^5.2.0",
    "loader.js": "^4.7.0",
    "qunit-dom": "^0.8.0",
    "sass": "^1.17.2"
  },
  "engines": {
    "node": "6.* || 8.* || >= 10.*"
  }
}
```

## Steps to reproduce (I will try to narrow it down tomorrow with a new project).

```shell
> ember -v
ember-cli: 3.7.1
node: 11.9.0
os: win32 x64
> ember new project
> ember install ember-cli-sass ember-concurrency ember-cli-mirage ember-moment ember-fetch ember-simple-auth
> ember serve
```

**Comment by LevelbossMike** [comment]
This is coming from ember data. Reference: https://github.com/emberjs/data/issues/5799

**Comment by stefanpenner** [comment]
These warnings are somewhat misleading, as these cycles are perfectly valid rollup is just newly reporting them.

I agree they are annoying, and we should likely silence them.  In-fact they have been silenced in ember-data: https://github.com/emberjs/data/pull/5839 and likely some near future release will make them go away.

In theory we could do this for all broccoli-rollup cycle warnings, by pushing the above mentioned PR into broccoli-roll up instead.

**Comment by stefanpenner** [comment]
Another option is to get roll up folks to make this configurable.

**Comment by rwjblue** [comment]
@stefanpenner rollup already allows the warning system to be pluggable

This issue was resolved by https://github.com/emberjs/data/pull/5839 and released in ember-data@3.8.0.
